components/Banner/examples/supplemental/MDCBannerView.h

```diff
@@ -18,15 +18,42 @@
 
 __attribute__((objc_subclassing_restricted)) @interface MDCBannerView : UIView
 
-@property(nonatomic, copy, nonnull) NSString *text;
+/**
+ A view that displays the text on a @c MDCBannerView.
+ @discussion Although this property is read-only, its properties are read/write. These properties can be used to configure the text shown on @c MDCBannerView.
+ For example:
+ @code
+ MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+ bannerView.textLabel.font = [UIFont systemFontOfSize: 12];
+ @endcode
+ */
+@property(nonatomic, readonly, strong, nonnull) UILabel *textLabel;
 
-@property(nonatomic, strong, nullable) UIImage *image;  // default is nil
+/**
+ A view that displays the image on a @c MDCBannerView.
+ @discussion Although this property is read-only, its properties are read/write. These properties can be used to configure the image shown on @c MDCBannerView.
+ For example:
+ @code
+ MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+ bannerView.imageView.image = [UIImage imageNamed:@"example-image"];
+ bannerView.imageView.tintColor = UIColor.whiteColor;
+ @endcode
+ If its own property @c image is @c nil, this view won't show on the @c MDCBannerView.
+ */
+@property(nonatomic, readonly, strong, nonnull) UIImageView *imageView;
 
-@property(nonatomic, assign) NSUInteger numberOfButtons;  // default is 1
+/**
+ A leading button that displays on a @c MDCBannerView.
+ @discussion This @c leadingButton displays on the leading edge of the view. If it can't occupy on the same line with @c trailingButton, it always shows above it.
+ The default value is nil.
+ */
+@property(nonatomic, readwrite, strong, nullable) MDCButton *leadingButton;
 
 /**
- The buttons representing the banner's actions.
+ A trailing button that displays on a @c MDCBannerView.
+ @discussion This @c trailingButton displays on the trailing edge of the view. If it can't occupy on the same line with @c trailingButton, it always shows below it.
+ The default value is nil.
  */
-@property(nonatomic, readonly, copy, nonnull) NSArray<MDCButton *> *buttons;
+@property(nonatomic, readwrite, strong, nullable) MDCButton *trailingButton;
```

Comment:
Same question here - why readwrite?  Do we want clients to assign an `MDCFloatingButton` or a custom subclass if they prefer?

Reply:
`trailingButton` is an optional button. If I make it readonly, once developer access it, it will be added to subview. And I need anther property to provide the functionality of hiding it, such as `trailingButtonHidden`. I expect people to set this property to a MDCButton when they need it, otherwise they can set it to `nil`. I agree this `readwrite` might not be optimal considering other properties are `readonly`. I can go with adding another `readwrite` property to control the visibility too.

Reply:
Talked offline. Addressed in c17c57d42ab6d485cc6570b2ccb6ded11981d656.